packages/zosjobs/__tests__/__unit__/MonitorJobs.unit.test.ts

```diff
@@ -286,22 +443,129 @@ describe("MonitorJobs", () => {
+                    try {
+                        await MonitorJobs.waitForOutputStatus(session, jobname, jobid);
+                    } catch (e) {
+                        error = e;
+                    }
+                    expect(error).toBeDefined();
+                    expect(error instanceof ImperativeError).toBe(true);
+                    expect(error.message).toMatchSnapshot();
+                });
+
+                it("should error if missing jobid", async () => {
+                    let error;
+                    const jobname = "FAKE";
+                    const jobid = undefined as any;
+                    try {
+                        await MonitorJobs.waitForOutputStatus(session, jobname, jobid);
+                    } catch (e) {
+                        error = e;
+                    }
+                    expect(error).toBeDefined();
+                    expect(error instanceof ImperativeError).toBe(true);
+                    expect(error.message).toMatchSnapshot();
+                });
+            });
+            describe("error handling", () => {
+                const parms: IMonitorJobWaitForParms = {
+                    jobid  : "FAKE",
+                    jobname: "FAKE"
+                };
+
+                it("should throw the proper error if catching an ImperativeError", async () => {
+                    const error = new ImperativeError({
+                        msg: "EXPECT THIS ERROR TO BE THROWN"
+                    });
+
+                    let expectError: any;
+
+                    waitForStatusCommonSpy.mockImplementationOnce(async () => {
+                        throw error;
+                    });
+
+                    try {
+                        await MonitorJobs.waitForStatusCommon(session, parms);
+                    } catch (e) {
+                        expectError = e;
+                    }
+
+                    expect(expectError).toBeInstanceOf(ImperativeError);
+                    expect(expectError.message).toMatchSnapshot();
+                });
+
+                it("should throw the proper error if catching a normal error", async () => {
+                    const error = new Error("THIS IS A NORMAL ERROR");
+
+                    let expectError: Error;
+
+                    waitForStatusCommonSpy.mockImplementationOnce(async () => {
+                        throw error;
+                    });
+
+                    try {
+                        await MonitorJobs.waitForStatusCommon(session, parms);
+                    } catch (e) {
+                        expectError = e;
+                    }
+
+                    expect(expectError).toBeInstanceOf(Error);
+                    expect(expectError.message).toMatchSnapshot();
+                });
 
-                const jobname = "JOB52712";
-                const jobid = "52712";
+                it("should throw the proper error if catching a wrapped error", async () => {
```

Comment:
I think this test is redundant with "should throw the proper error if catching an ImperativeError", since a "wrapped error" means the same thing as an "Imperative error".

Reply:
will remove wrapped test